### ISSUE TITLE
SC2151 — False positive

### ISSUE THREAD

**Issue by effleurager** [koalaman/shellcheck#2599]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2151
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
a(){
        return [[ -f "$0" ]]
}

a
```

#### Here's what shellcheck currently says:
```
Line 3	SC2151: Only one integer 0-255 can be returned. Use stdout for other data.
```
#### Here's what I wanted or expected to see:

```
```

The return values of tests always comply with the limitations for singular integer between 0 and 255. This is a false-positive issue.

**Comment by emanuele6** [comment]
Your code is not correct, and if you run it, you will get an error, because it is not correct. `bash: return: [[: numeric argument required`

The correct way to write that is `[[ -f "$0" ]]; return`.
